datascience/table.py

```diff
@@ -320,22 +320,30 @@ def groups(self, column_labels, collect=lambda s: s):
         return grouped
 
     def pivot(self, columns, rows, values, collect=lambda s:s, zero=None):
-        """Generate a table with a column for rows and a column for each
-        unique value in columns. Each row aggregates over the values that
-        match both row and column.
+        """Generate a table with a column for rows (or a column for each row
+        in rows list) and a column for each unique value in columns. Each row
+        aggregates over the values that match both row and column.
 
         columns, rows, values -- column labels in self
         collect -- aggregation function over values
         zero -- zero value for non-existent row-column combinations
         """
-        selected = self.select([columns, rows, values])
-        grouped = selected.groups([columns, rows], collect)
-        row_values = np.unique(self._get_column(rows))
-        pivoted = Table([row_values], [rows])
+        if not _is_non_string_iterable(rows):
+            rows = [rows]
+        selected = self.select([columns, values] + rows)
+        grouped = selected.groups([columns] + rows, collect)
+
+        # Generate all possible combinations of values from columns in rows
+        rows_values = \
+            list(itertools.product(*(np.unique(self._get_column(column_label))
```

Comment:
I think we actually want only sets for which there exists at least one row, rather than the `product` of all possible combos. So, `np.unique` the tuples of row values.